Mathlib/RingTheory/KrullDimension.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li, Jujian Zhang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+import Mathlib.Topology.KrullDimension
+
+/-!
+# Krull dimensions of (commutative) rings
+
+Given a commutative ring, its ring theoretic Krull dimension is the order theoretic Krull dimension
+applied to its prime spectrum. Unfolding this definition, it is the length of the longest series of
+prime ideals ordered by inclusion.
+
+
+## Main definitions and results
+
+* `ringKrullDim`: the ring theoretic Krull dimension of a commutative ring.
+
+* `ringKrullDim.eq_topologicalKrullDim`: the ring theoretic Krull dimension of a commutative ring
+is equal to the topological Krull dimension of its prime spectrum.
+
+* `ringKrullDim.eq_zero_of_field`: the Krull dimension of a field is zero.
+-/
+
+/--
+The ring theoretic Krull dimension is the Krull dimension of its spectrum ordered by inclusion.
+-/
+noncomputable abbrev ringKrullDim (R : Type*) [CommRing R] : WithBot (WithTop ℕ) :=
+  krullDim (PrimeSpectrum R)
+
+namespace ringKrullDim
+
+lemma eq_bot_of_subsingleton (R : Type*) [CommRing R] [Subsingleton R] :
+    ringKrullDim R = ⊥ :=
+  krullDim_eq_bot_of_isEmpty
+
+lemma nonneg_of_nontrivial (R : Type*) [CommRing R] [Nontrivial R] :
+    0 ≤ ringKrullDim R :=
+  krullDim_nonneg_of_nonempty
+
+theorem eq_topologicalKrullDim (R : Type*) [CommRing R] :
+    ringKrullDim R = topologicalKrullDim (PrimeSpectrum R) :=
+  Eq.symm $ krullDim_orderDual.symm.trans $ krullDim_eq_of_orderIso
+  (PrimeSpectrum.pointsEquivIrreducibleCloseds R).symm
+
+/-- If `f : R →+* S` is surjective, then `ringKrullDim S ≤ ringKrullDim R`. -/
+theorem le_of_surjective {R S : Type*} [CommRing R] [CommRing S] (f : R →+* S)
+    (hf : Function.Surjective f) : ringKrullDim S ≤ ringKrullDim R :=
+  krullDim_le_of_strictMono (PrimeSpectrum.comap f) (Monotone.strictMono_of_injective
+    (fun _ _ hab ↦ Ideal.comap_mono hab) (PrimeSpectrum.comap_injective_of_surjective f hf))
+
+/-- If `I` is an ideal of `R`, then `ringKrullDim (R ⧸ I) ≤ ringKrullDim R`. -/
+theorem quotient_le {R : Type*} [CommRing R] (I : Ideal R) :
+    ringKrullDim (R ⧸ I) ≤ ringKrullDim R :=
+  le_of_surjective _ Ideal.Quotient.mk_surjective
+
+/-- If `R` and `S` are isomorphic, then `ringKrullDim R = ringKrullDim S`. -/
+theorem eq_of_ringEquiv {R S : Type*} [CommRing R] [CommRing S] (e : R ≃+* S) :
```

Comment:
I would give this an alias `RingEquiv.ringKrullDim` to enable dot notation.

Reply:
I have written a new theorem `RingEquiv.ringKrullDim`, whose statement is exactly the same as that of `ringKrullDim_eq_of_ringEquiv`. Is it the correct way of creating an alias?

Reply:
Could you please have a look and see if I have created the alias properly (i.e. have I put the alias in the correct place, etc.)? Thanks a lot!